File: src/main/java/ru/spbstu/spartamonitor/listeners/DrawDensityListener.java
package ru.spbstu.spartamonitor.listeners;

import com.google.common.eventbus.Subscribe;
import ru.spbstu.spartamonitor.SpartaMonitorController;
import ru.spbstu.spartamonitor.events.DrawDensityEvent;

public record DrawDensityListener(SpartaMonitorController controller) {
    @Subscribe
    public void handleDrawEvent(DrawDensityEvent event) {
        controller.drawDensityChart(event.xCoord());
    }
}

File: src/main/kotlin/ru/spbstu/spartamonitor/listeners/DrawDensityListener.kt
package ru.spbstu.spartamonitor.listeners

import com.google.common.eventbus.Subscribe
import ru.spbstu.spartamonitor.SpartaMonitorController
import ru.spbstu.spartamonitor.events.DrawDensityEvent

data class DrawDensityListener(val controller: SpartaMonitorController) {
    @Subscribe
    fun handleDrawEvent(event: DrawDensityEvent) {
        controller.drawDensityChart(event.xCoord)
    }
}
